src/controllers/authenticationController.ts

```diff
@@ -104,6 +106,14 @@ export default class AuthenticationController {
 
     const spidUser = errorOrSpidUser.value;
 
+    if (!isOlderThan(14)(parse(spidUser.dateOfBirth), new Date())) {
```

Comment:
what if `dateOfBirth` is undefined?

Reply:
Do we have to fallback calculating `dateOfBirth` from fiscal code?

Reply:
`dateOfBirth` should be always present, in #831 we have added this param to the required attributes.
Nevertheless if for some reason the `dateOfBirth` will be missing this code now is not executed because the `SpidUser.decode` will fail  ([here](https://github.com/pagopa/io-backend/pull/844/files#diff-08f03361c8ba099431787cddb0e1fece15ddf1939e33968cc455ede3d22fdf79R97) the link to the type change in this PR.)

Reply:
I wonder if `dateOfBirth` can be an empty string, but if it is not possible, LGTM :)

Reply:
@AleDore what if we add a strict typings for the `dateOfBirth` field in `SpidUser` decoder?
```typescript
export const SpidDateOfBirth = PatternString("^[0-9]{4}-[0-9]{2}-[0-9]{2}$");
export type SpidDateOfBirth = t.TypeOf<typeof SpidDateOfBirth>;

// required attributes
export const SpidUser = t.intersection([
  t.interface({
    authnContextClassRef: SpidLevel,
    dateOfBirth: SpidDateOfBirth,
    familyName: t.string,
    fiscalNumber: FiscalCode,
    getAssertionXml: t.Function,
    issuer: Issuer,
    name: t.string
  }),
  t.partial({
    email: EmailAddress,
    nameID: t.string,
    nameIDFormat: t.string,
    sessionIndex: t.string
  })
]);
```

Reply:
seems legit :)

Reply:
Great, but i guess it should be a `Date` not a `PatternString`, what about it? (see `DateFromString`)

Reply:
Change the type from `string` to `Date` require a refactory of `toAppUser` method and the eventually the types redis related (es. UserWithoutTokens) where the type of the field `date_of_birth` is now a simple string.
Reading the code maybe a side effect could happen for dates with the format `YYYY-MM-G` (look [here](https://github.com/pagopa/io-backend/blob/IC-109-login-age-limit/src/types/user.ts#L131)) when the `DateFromString` generate wrong dates. From SPID the dates should always have the format `YYYY-MM-GG` ([ref](https://docs.italia.it/italia/spid/spid-regole-tecniche/it/stabile/attributi.html)) but i'm not sure that is the same for CIE.

Reply:
I confirm that is a CIE problem. The fix was added by #699

Reply:
I guess that behavior is due to the fact that the object isn't passed to `encode()` (which would convert the date to a string again) before putting it into redis.

Reply:
If you agree I would prefer to minimize the changes in this release. All this improvements could be done on another PR improving the typings of `SpidUser` and redis-related objects.

Reply:
Yes, of course!